src/bfs_tree.c

```diff
@@ -77,106 +174,122 @@ BFSNode *bfs_tree_create_node(char fname[BFS_MAX_NAME_LENGTH]) {
     return node;
 }
 
-BFSNode *bfs_tree_create_path(BFSTree root, char path[BFS_PATH_SIZE]) {
-    if(root == NULL) {
-        return EMPTY_NODE;
+int bfs_tree_search(BFSNode *root, char *s) {
+    if(check_pointer(root)) {
+        return 0;
+    }    
+    
+    if(strcmp(root->fname, s) == 0) {    
+        return 1;
     }
 
-    unsigned int i = 1;
+    return bfs_tree_search(root->child, s) + bfs_tree_search(root->next, s);
+}
 
-    // NULL subtree
-    if(root->child == NULL) {
-        BFSNode *walker = root;
+int bfs_tree_search_child(BFSNode *child, char *s) {
+    if(check_pointer(child)) {
+        return 0;
+    }
 
-        // Create all the nodes of the given path
-        while(i < bfs_count_delim(path, '/')) {
-            BFSNode *node = bfs_tree_create_node(bfs_strsplit(path, "/", i));
-            walker->child = node;
-            node->father = walker;
-            walker = walker->child;
-            i++;
-        }
+    if(strcmp(child->fname, s) == 0) {    
+        return 1;
+    }
+    
+    return bfs_tree_search_child(child->next, s);
+}
+
+BFSNode *tmp;
 
-        return walker;
+BFSNode *bfs_tree_get_child(BFSNode *root, char *s) {
+    if(check_pointer(root)) {
+        return EMPTY_NODE;
+    }
+    else if(strcmp(root->fname, s) == 0) {
+        tmp = root;
     }
     else {
-        BFSNode *walker = root->child;
+        bfs_tree_get_child(root->child, s);
+        bfs_tree_get_child(root->next, s);
+    }
+    return tmp;
+}
 
-        while(walker->next != NULL &&
-              bfs_strsplit(path, "/", i) != walker->fname) {
-            walker = walker->next;
-        }
+void bfs_tree_print(BFSTree root, char *arg) {
+    if(root == NULL) {
+        return;
+    }
+    else if(root->child == NULL && root->fpermissions == BFS_NODE_IS_DIR) {
+        bfs_pferror(stderr, "Empty directory", __LINE__, __FILE__,
```

Comment:
This shouldn't be printed as a warning. We certainly have to open an issue for this.